File: 20101020 - Espiral - ruby/espiral.rb
def gera_espiral (linhas, colunas)
  if linhas*colunas == 4
    (1..colunas).to_a.join(' ')+(colunas+1..linhas*colunas).to_a.join(' ')
  elsif linhas > 1
    (1..linhas).to_a.join("\n")
  else
    (1..colunas).to_a.join(' ')
  end
end

File: 20101027 - espiral - python/espiral.py
def foo(colunas, espacador, linhas=1):
    numeros = range(linhas, colunas + linhas)
    tamanho = len(str(colunas))
    return espacador.join([str(numero).rjust(tamanho) for numero in numeros])


def espiral(linhas, colunas):
    if linhas == colunas == 2:
        return foo(colunas, ' ', 1) + '\n' + foo(colunas, ' ', 3)[::-1]

    if linhas > 1:
        return foo(linhas, '\n')
    else:
        return foo(colunas, ' ')

